opmd_viewer/openpmd_timeseries/interactive.py

```diff
@@ -274,24 +294,26 @@ def step_bw(b) :
             # Species selection
             ptcl_species_button = widgets.Dropdown( width=250,
                 options=self.avail_species )
-            ptcl_species_button.on_trait_change( refresh_ptcl )
-            # Remove charge and mass (less interesting) 
-            avail_ptcl_quantities = [ q for q in self.avail_ptcl_quantities \
+            ptcl_species_button.on_trait_change( refresh_species )
+            # Get available records for this species (remove charge and
+            # mass as they are typically less interesting)
+            avail_records = [ q for q in \
+                        self.avail_record_components[ptcl_species_button.value]
                         if (q in ['charge', 'mass'])==False ]
             # Particle quantity on the x axis
             ptcl_xaxis_button = widgets.ToggleButtons(
-                value='z', options=avail_ptcl_quantities )
```

Comment:
maybe we can create the buttons with an empty list here and fill them with a `refresh` at the end.

that would reduce code duplication and separate layouting from data filling.